docs/references/keywords.rst

```diff
@@ -194,7 +194,46 @@ extensions).
 .. _keyword/package_dir:
 
 ``package_dir``
-    A dictionary providing a mapping of package to directory names.
+    A dictionary that maps package names (as the developer wishes they would be
+    imported by the end-users) into directory paths (that actually exist in the
+    project's source tree). This configuration can be used with 2 main purposes:
+
+    1. To effectively "rename" paths when building your package.
+       For example, ``package_dir={"mypkg": "dir1/dir2/code_for_mypkg"}``
+       will instruct setuptools to copy the ``dir1/dir2/code_for_mypkg/...`` files
+       as ``mypkg/...`` when building the final :term:`wheel distribution <Wheel>`.
+
+       .. attention::
+          While it is *possible* to specify arbitrary mappings, developers are
+          **STRONGLY ADVISED AGAINST** that. They should try as much as possible
+          to keep the directory names and hierarchy identical to as they would
+          appear in the final wheel, only deviating when absolutely necessary.
+
+    2. To indicate that the relevant code is entirely contained inside
+       a specific directory (instead of directly placed under the project's root).
+       In this case, a special key is required (the empty string, ``""``),
+       for example: ``package_dir={"": "<name of the container directory>"}``.
+       All the directories inside the container directory will be copied
+       directly into the final :term:`wheel distribution <Wheel>`, but the
+       container directory itself will not.
+
+       This practice is very common in the community, to help separating the
+       package implementation from auxiliary files (e.g. CI configuration files),
+       and is referred as :ref:`src-layout`, because traditionally the
```

Comment:
```suggestion
       and is referred to as :ref:`src-layout`, because traditionally the
```
Would "commonly" be better than "traditionally"?

Reply:
I could go with both...

Naming the directory `src` is part of a long-established practice, so that is why I used the "traditionally" word here (is it wrong to call this long-established practice a tradition?)

Reply:
I see you've already made the change; but in any case, 'traditionally' carries the connotation that it might not be current or recommended practice, hence my suggestion that 'commonly' is better.

Reply:
Thank you very much @cdfarrow, it is always nice to have this kind of feedback. It helps a lot (I am not a native speaker...)

Reply:
I'm glad to be able to contribute a little. Thanks for your quick response to my questions, and for adding the clarifications. (I've also learned a bit more about using Github in the process :-) )